src/clients/bridge.ts

```diff
@@ -0,0 +1,71 @@
+import { TextChannel } from 'discord.js';
+import { linkedListNode, nodeInterface } from '../linkedList';
+import { twitchMsg } from '../messageObjects';
+import tmijs from 'tmi.js';
+import configFile from '../../config.json';
+
+/**
+ * If an error happens either on twitch or discord, print the thing
+ * @param {*} error 
+ */
+export const genericPromiseError = (error: any) => console.error('Snap, I hit a snag... >.<', error);
+
+interface IBridge
+{
+    MAX_MSG_CACHE: number;
+    currMsgCount: number;
+    targetDiscordChannel: TextChannel | undefined,
+    twitchClient: tmijs.Client | undefined,
+    discordTwitchCacheMap: Map<any, any>
+    twitchMessageSearchCache: { [key: string]: linkedListNode; };
+    messageLinkdListInterface: nodeInterface;
+    lastUserStateMsg: any
+}
+
+const Bridge: IBridge = {
+    MAX_MSG_CACHE: 100,
+    currMsgCount: 0,
+    targetDiscordChannel: undefined,
+    twitchClient: undefined,
+    discordTwitchCacheMap: new Map(),
+    twitchMessageSearchCache: {},
+    messageLinkdListInterface: new nodeInterface(),
+    lastUserStateMsg: null
+}
+
+
+function manageMsgCache(specificNode?: linkedListNode): number | linkedListNode
```

Comment:
Wait... it can return a number? 🥴 

ahhh wait; ok I see now. I don't think it will return a number, but it can return `null`

if we get *reeeeealy* technical, `0 == null`, but if it's typescript we're dealing with, might be more clear to just return a `null` type

Reply:
![image](https://user-images.githubusercontent.com/108752129/187559286-78a862f4-b5f3-48bd-bd63-964ab0a24f58.png)
Nevermind; we don't have to worry about this until we go about ironing out the actual program

The task of the branch was to re-organize where things were, so we can ignore for now